### PR TITLE
[Analyzer] Build scoped pods once for each target definition

### LINKED ISSUE #4034: Target Deduplication removes necessary Pods project targets 🌈
I ran into this issue when upgrading from CocoaPods `0.37.2` to `0.38.0`, and it's not fixed as of `0.38.2`.

This issue is a bit long, but only because I wanted to make it very clear what is happening -- the [sample project](https://github.com/smileyborg/CPTargetDeduplication) should make this very easy to understand.

## Background
I ran into this issue with the QuickBooks app I work on, which has an unusually complex build configuration. Aside from CocoaPods, we have a workspace of multiple separate Xcode projects, and each Xcode project generally has 2 framework targets -- one for iOS and one for Mac. This is because we use these independent Xcode projects (and their frameworks) to share code between multiple iOS and OS X apps. We're also using CocoaPods to manage our third-party dependencies.

I've created a [small git repo](https://github.com/smileyborg/CPTargetDeduplication) that contains a sample workspace that represents a simple version of the structure of our real QuickBooks workspace. In this git repo:

* `ProjectABWorkspace`
  * `ProjectA` (has a dependency on `Pods`)
    * `ProjectA-iOS`
    * `ProjectA-Mac`
  * `ProjectB` (has a dependency on `ProjectA`, and on `Pods`)
    * `ProjectB-iOS`
    * `ProjectB-Mac`
  * `Pods`

As you can see, there are 2 separate Xcode projects and a Pods project (generated by CocoaPods). Each of the `ProjectA` and `ProjectB` projects has 2 framework targets it produces, one for iOS and one for OS X. Both `ProjectA` and `ProjectB` depend on a CocoaPod (INTUGroupedArray). And finally, `ProjectB` depends on `ProjectA`, by embedding & linking the `ProjectA.framework`.

The [Podfile in this repo](https://github.com/smileyborg/CPTargetDeduplication/blob/master/Podfile) has a separate Pod target for each framework target, making 4 total Pod targets.

## The Issue
With `0.37.2` all of the framework targets build fine. (You can checkout [this commit](https://github.com/smileyborg/CPTargetDeduplication/commit/dda3a6cdf5b6fe732c5fa390728e8dffac1694a5) to see that.)

However, with `0.38.0` or higher, both of the ProjectB framework targets fail to build (use the `ProjectB-iOS` and `ProjectB-Mac` schemes). The build failure itself is within the `ProjectBFile.m` import to the INTUGroupedArray CocoaPod.

The actual issue here is more clear when you look at the git log for the repo. If you look at the diff for [the commit that runs `pod install` using `0.38.2`](https://github.com/smileyborg/CPTargetDeduplication/commit/bbd8d1f16905aa653e7e2ee31326c5a561767d0f) (after already having run `pod install` using `0.37.2`), you can see two significant sets of deletions:
* All the files in `Pods/Target Support Files/Pods-ProjectB-Mac-INTUGroupedArray/` have been removed
* All the files in `Pods/Target Support Files/Pods-ProjectB-iOS-INTUGroupedArray/` have been removed

If you look at the Pods project itself at this point, it is also missing the `Pods-ProjectB-iOS-INTUGroupedArray` and `Pods-ProjectB-iOS-INTUGroupedArray` targets. So the root cause of the compilation error is that the INTUGroupedArray pod is no longer being linked with the ProjectB targets.

CocoaPods `0.38.0` introduced [target deduplication](http://blog.cocoapods.org/CocoaPods-0.38/#target-deduplication), which appears to be causing this issue. It looks to me like the *Target Deduplicator* decided these were not necessary (even though they were) and removed them.

This issue is blocking us from upgrading beyond CocoaPods `0.37.2`.

Let me know if you have any questions or want more information!

### PR DESCRIPTION
**Pull request by mrackwitz**
Fixes #4034. Alternative approach to #4146.

#### ToDo:

* [x] Changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Either generate just one pod target or generate it once for each target definition.
```
Comment (segiddins) [inline CHANGELOG.md]: wrong section

**Comment by segiddins** [comment]
:+1: if we can verify that this indeed fixes things

File: CHANGELOG.md (hunk 1 above)
Comment (mrackwitz) [inline CHANGELOG.md]: but just wrong version meanwhile?

Comment (segiddins) [inline CHANGELOG.md]: ah yeah

**Comment by segiddins** [comment]
I've now rebased this, it's good to merge once CI passes (I manually verified that it fixed the linked issue).

**Comment by segiddins** [comment]
:fireworks: thanks marius!

**Comment by mrackwitz** [comment]
Thanks for taking over the last mile, Sam!

**Pull request merged**